### PR TITLE
add comments

### PR DESCRIPTION
**Pull request by belforte**
add a few comments to `.gitlab-ci.yaml`

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -28,7 +30,9 @@ stages:
   - tagging_release
 
 get_env:
+  # NB if there is no rule the job runs all of the times
   rules:
+    # !reference in following lines tells GITLAB to replace with the value of the indicated varaibles
```
Comment (novicecpp) [inline .gitlab-ci.yml:35]:
```suggestion
    # !reference in following lines tells GITLAB to replace with the value of the indicated variables.
```

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -12,6 +13,7 @@ variables:
   DOCKER_TLS_CERTDIR: "/certs"
   DOCKER_BUILDKIT: 1
 
+# this defines two variables .default_rules.default and .default_rules.release which are True when the if is satisfied
```
Comment (novicecpp) [inline .gitlab-ci.yml:16]:
```suggestion
This key define variables which later use in `!reference` tag in `rules`. 
# Ref https://docs.gitlab.com/ee/ci/jobs/index.html#hide-jobs 
# and https://docs.gitlab.com/ee/ci/yaml/yaml_optimization.html#reference-tags
```

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -28,7 +30,9 @@ stages:
   - tagging_release
 
 get_env:
+  # NB if there is no rule the job runs all of the times
```
Comment (novicecpp) [inline .gitlab-ci.yml:33]:
```suggestion
  # NB rules are evaluated like python `any()`. If there is no rule the job runs all of the times. 
```

**Review by novicecpp — COMMENTED** [review]
I modified it a bit. What do you think?

File: .gitlab-ci.yml (hunk 1 above)
Comment (belforte) [reply .gitlab-ci.yml:35]: sorry for the typo

File: .gitlab-ci.yml (hunk 2 above)
Comment (belforte) [reply .gitlab-ci.yml:16]: OK. Please change "which later" to "which are later used" . (avoid a verb w/o a subject)

Comment (belforte) [reply .gitlab-ci.yml:16]: I do not seem able to edit your commit

Comment (novicecpp) [reply .gitlab-ci.yml:16]: Done

**Comment by belforte** [comment]
thanks @novicecpp

**Pull request merged**